validator/client/validator.go

```diff
@@ -113,16 +113,28 @@ func (v *validator) UpdateAssignments(ctx context.Context, slot uint64) error {
 		PublicKey:  v.key.PublicKey.Marshal(),
 	}
 
-	resp, err := v.validatorClient.ValidatorEpochAssignments(ctx, req)
+	resp, err := v.validatorClient.CommitteeAssignment(ctx, req)
 	if err != nil {
 		return err
 	}
 
-	v.assignment = resp.Assignment
+	v.assignment = resp
+
+	var proposerSlot uint64
+	var attesterSlot uint64
+	if v.assignment.IsProposer && len(v.assignment.Committee) == 1 {
```

Comment:
You can only be an attester AND proposer if committee size is 1?

Reply:
that's correct, because the first index is proposer. More I think about this, we can just simply it to `if len(v.assignment.Committee) == 1`

Reply:
Can you add some comments to the code? Explain why this can only be the case. Thanks